src/main/java/software/amazon/cloudwatchlogs/emf/model/DimensionSet.java

```diff
@@ -149,6 +151,10 @@ private static DimensionEntry entryOf(String key, String value) {
      * @param value Value of the dimension
      */
     public void addDimension(String dimension, String value) {
+        if (this.getDimensionKeys().size() == Constants.MAX_DIMENSIONS) {
```

Comment:
Can we use >= to be safer here?
Once we start throwing it shouldn't be possible to go beyond the limit, but in case there's a way that it could we would still catch it here.